File: 271-encode-and-decode-strings/encode-and-decode-strings.java
public class Codec {

    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
        String res = "";
        for (String s : strs) {
            res += s.length() + "#" + s;
        }

        return res;
    }

    // Decodes a single string to a list of strings.
    public List<String> decode(String s) {
        int l = 0;
        List<String> res = new ArrayList<>();
        while (l < s.length()) {
            int r = l;
            while (s.charAt(r) != '#') {
                r += 1;
            }

            int length = Integer.parseInt(s.substring(l, r));
            int start = r + 1;
            res.add(s.substring(start, start + length));

            l = start + length;
        }

        return res;
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(strs));

File: 271-encode-and-decode-strings/encode-and-decode-strings.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        if not strs:
            return ""
        
        res = ""
        for s in strs:
            res += str(len(s)) + "#" + s
        
        return res

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """
        if not s:
            return []

        res = []
        i = 0
        while i < len(s):
            length = 0
            while s[i] != "#":
                length = length * 10 + int(s[i])
                i += 1
            i += 1
            word = s[i: i + length]
            res.append(word)
            i += length
        
        return res

# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))